client/elements/text/sc-segmented-text.html

```diff
@@ -550,22 +579,179 @@ <h3>Could not find the sutta text.</h3>
               return ((!error && loading) || isTextViewHidden);
           }
 
+          _getElememtToHighlight(element) {
+              if (element.id.includes('inline') && !element.id.includes('.')) {
+                  element = element.parentNode;
+              }
+              return element.parentNode;
+          }
+
+          _generateId(element) {
+              if (element && element.includes('.') && !element.includes('inline')) {
+                  return `${element}_inline`;
+              }
+              return element;
+          }
+
+          _gotScSeg(wrapper) {
+              return wrapper.querySelectorAll('sc-seg').length === 1;
+          }
+
+          _removeHighlights() {
+              this.shadowRoot.querySelectorAll('.highlight').forEach(v => {
+                  v.classList.remove('highlight');
+                  v.classList.remove('last-highlight');
+              })
+              this.shadowRoot.querySelectorAll('.highlight-wrapper-highlighted').forEach(v => {
+                  v.classList.remove('highlight-wrapper-highlighted');
+              })
+          }
+
+          _disableHighlightWrappers() {
+              this.shadowRoot.querySelectorAll('.highlight-wrapper').forEach(v => {
+                  v.classList.remove('highlight-wrapper');
+                  v.classList.add('disabled-highlight-wrapper');
+              })
+          }
+
+
+          _putSectionIntoWrapper(section) {
+              let wrapperAlreadyExists = false;
+              let wrapper;
+              if (section.parentNode && section.parentNode.classList.contains('highlight-wrapper')) {
+                  section.parentNode.classList.add('highlight-wrapper-highlighted');
+                  wrapperAlreadyExists = true;
+              } else {
+                  if (section.previousElementSibling && (section.nodeName !== 'SC-SEG' || (
+                          this._gotScSeg(section.previousElementSibling) &&
+                          section.previousElementSibling.classList.contains('highlight-wrapper-highlighted')))) {
+                      wrapper = section.previousElementSibling;
+                  } else {
+                      wrapper = document.createElement('span');
+                  }
+                  wrapper.classList.add('highlight-wrapper-highlighted');
+                  wrapper.classList.add('highlight-wrapper');
+                  section.parentNode.insertBefore(wrapper, section);
+                  wrapper.appendChild(section.cloneNode(true));
+              }
+              return [wrapperAlreadyExists, wrapper];
```

Comment:
It seems to me that if `wrapperAlreadyExists`, then `wrapper` is always null, and when `wrapper` is not null, `wrapperAlreadyExists` is always false. If that's correct, I don't see how this function needs to return two variables. 

The better pattern here (consistent with the Single Responsibility Principle) would be to make a separate function that checks if it exists and call that independently. This would also reduce conditional nesting.